Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandItemsStackSplit.java

```diff
@@ -23,61 +28,93 @@ public CommandItemsStackSplit(ODPDBAccess db, HttpServletRequest request, HttpSe
 	protected void processBatchItems(Map<String, String> parameters, ODPDBAccess db, CachedDatastoreService ds,
 			CachedEntity character, List<CachedEntity> batchItems) throws UserErrorMessage {
 		Long splitItemQuantity;
-		if (batchItems.size() > 1) {
-			throw new UserErrorMessage("You can only split one item at a time.");
-		}
+		if (batchItems.size() > 12) 
+			throw new UserErrorMessage("You can only split 12 items at a time.");
+		
 		ds.beginTransaction();
 		try {
-			CachedEntity splitItem = batchItems.get(0);
-			
-			if (CommonChecks.isItemCustom(splitItem)) 
-				throw new UserErrorMessage("You cannot split a custom item.");
-			
-			splitItem = db.getDB().refetch(splitItem);
-			
-			if (GameUtils.equals(splitItem.getProperty("containerKey"), character.getKey()) == false) {
-				throw new UserErrorMessage("Item does not belong to character.");
-			}
-			if (!splitItem.hasProperty("quantity")) {
-				throw new UserErrorMessage("You can only split stackable items.");
-			}
-			splitItemQuantity = (Long) splitItem.getProperty("quantity");
-			if (splitItemQuantity==null){
-				throw new UserErrorMessage("You can only split stackable items.");
-			}
-			if (splitItemQuantity < 2) {
-				throw new UserErrorMessage("You can only split stacks of two or more items.");
-			}
 			
 			Long stackSize;
+			Long stacks;
 			try {
 				stackSize = tryParseId(parameters, "stackSize");
-			} catch (NumberFormatException e) {
+				stacks = tryParseId(parameters, "stacks");
+			}
+			catch (NumberFormatException e) {
 				throw new UserErrorMessage("Please input an integer value.");
 			}
-			if (stackSize <= 0) {
+			
+			if (stackSize <= 0 || stacks <= 0) 
 				throw new UserErrorMessage("Please input a positive integer value.");
+			
+			if(batchItems.size() > 1 && stacks > 1) 
+				throw new UserErrorMessage("You can't split multiple items into multiple stacks.");
+			
+			if(stacks > 24) 
+				throw new UserErrorMessage("Too many stacks. Max is 24.");
+			
```

Comment:
Think about how many items you are creating if you have 10 items of 100 quantity each making 10 stacks of 10.